docs/source/features/apq.md

```diff
@@ -0,0 +1,170 @@
+This example persists a dummy query of `{__typename}`, using its sha256 hash: `ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38`.
+
+
+1. Request a persisted query:
+
+```bash
+curl -g 'http://localhost:4000/?extensions={"persistedQuery":{"version":1,"sha256Hash":"ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38"}}'
+```
+
+   Expect a response of: `{"errors": [{"message": "PersistedQueryNotFound", "extensions": {...}}]}`.
+
+2. Store the query to the cache:
+
+```bash
+curl -g 'http://localhost:4000/?query={__typename}&extensions={"persistedQuery":{"version":1,"sha256Hash":"ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38"}}'
+```
+
+   Expect a response of `{"data": {"__typename": "Query"}}"`.
+
+3. Request the persisted query again:
+
+```bash
+curl -g 'http://localhost:4000/?extensions={"persistedQuery":{"version":1,"sha256Hash":"ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38"}}'
+```
+
+   Expect a response of `{"data": {"__typename": "Query"}}"`, as the query string is loaded from the cache.
+
+<h3 id="get">Using `GET` requests with APQ on a CDN</h3>
+
+A great application for APQ is running Apollo Server behind a CDN. Many CDNs only cache GET requests, but many GraphQL queries are too long to fit comfortably in a cacheable GET request.  When the APQ link is created with `createPersistedQueryLink({useGETForHashedQueries: true})`, Apollo Client automatically sends the short hashed queries as GET requests allowing a CDN to serve those request. For full-length queries and for all mutations, Apollo Client will continue to use POST requests.
+
+<h3 id="how-it-works">How it works</h3>
+
+The mechanism is based on a lightweight protocol extension between Apollo Client and Apollo Server. It works as follows:
+
+- When the client makes a query, it will optimistically send a short (64-byte) cryptographic hash instead of the full query text.
+- **Optimized Path:** If a request containing a persisted query hash is detected, Apollo Server will look it up to find a corresponding query in its registry. Upon finding a match, Apollo Server will expand the request with the full text of the query and execute it.
+
+  <img src="../images/persistedQueries.optPath.png" width="80%" style="margin: 5%" alt="Optimized Path">
+
+- **New Query Path:** In the unlikely event that the query is not already in the Apollo Server registry (this only happens the very first time that Apollo Server sees a query), it will ask the client to resend the request using the full text of the query. At that point Apollo Server will store the query / hash mapping in the registry for all subsequent requests to benefit from.
+
+<img src="../images/persistedQueries.newPath.png" width="80%" style="margin: 5%;" alt="New Query Path">
+
+<h2 id="cdn">CDN Integration</h2>
```

Comment:
I think you already mentioned this in real life: this section feels redundant with the CDN integration section below.